lib/luacrypto_skcipher.c

```diff
@@ -0,0 +1,240 @@
+* Plaintext length should be appropriate for the cipher mode (e.g., multiple of blocksize).
+* @function crypto_skcipher:encrypt
+* @tparam string iv The Initialization Vector. Its length must match `ivsize()`.
+* @tparam string plaintext The data to encrypt.
+* @treturn string The ciphertext.
+* @raise Error on encryption failure, incorrect IV length, or allocation issues.
+*/
+LUACRYPTO_SKCIPHER_CRYPT_FN(encrypt);
+
+/***
+* Decrypts ciphertext using the SKCIPHER transform.
+* The IV must match the one used during encryption.
+* Ciphertext length should be appropriate for the cipher mode.
+* @function crypto_skcipher:decrypt
+* @tparam string iv The Initialization Vector. Its length must match `ivsize()`.
+* @tparam string ciphertext The data to decrypt.
+* @treturn string The plaintext.
+* @raise Error on decryption failure, incorrect IV length, or allocation issues.
+*/
+LUACRYPTO_SKCIPHER_CRYPT_FN(decrypt);
+
+/***
+* Lua C methods for the SKCIPHER TFM object.
+* Includes cryptographic operations and Lunatik metamethods.
+* The `__close` method is important for explicit resource cleanup.
+* @see crypto_skcipher
+* @see lunatik_closeobject
+*/
+
+static const luaL_Reg luacrypto_skcipher_mt[] = {
+	{"setkey", luacrypto_skcipher_setkey},
+	{"ivsize", luacrypto_skcipher_ivsize},
+	{"blocksize", luacrypto_skcipher_blocksize},
+	{"encrypt", luacrypto_skcipher_encrypt},
+	{"decrypt", luacrypto_skcipher_decrypt},
+	{"__gc", lunatik_deleteobject},
+	{"__close", lunatik_closeobject},
+	{"__index", lunatik_monitorobject},
+	{NULL, NULL}
+};
+
+/***
+* Lunatik class definition for SKCIPHER TFM objects.
+* This structure binds the C implementation (luacrypto_skcipher_t, methods, release function)
+* to the Lua object system managed by Lunatik.
+*/
+
+static const lunatik_class_t luacrypto_skcipher_class = {
+	.name = "crypto_skcipher",
+	.methods = luacrypto_skcipher_mt,
+	.release = luacrypto_skcipher_release,
+	.sleep = true,
+	.pointer = true,
+};
+
+/***
+* Creates a new SKCIPHER transform (TFM) object.
+* This is the constructor function for the `crypto_skcipher` module.
+* @function crypto_skcipher.new
+* @tparam string algname The name of the skcipher algorithm (e.g., "cbc(aes)", "ctr(aes)").
+* @treturn crypto_skcipher The new SKCIPHER TFM object.
+* @raise Error if the TFM object or kernel request cannot be allocated/initialized.
+* @usage
+*   local skcipher_mod = require("crypto_skcipher")
+*   local cipher = skcipher_mod.new("cbc(aes)")
+*/
+static int luacrypto_skcipher_new(lua_State *L) {
+	const char *algname = luaL_checkstring(L, 1);
+	struct crypto_skcipher *tfm;
+	lunatik_object_t *object = lunatik_newobject(L, &luacrypto_skcipher_class, 0);
+
+	tfm = crypto_alloc_skcipher(algname, 0, 0);
+	if (IS_ERR(tfm)) {
+		long err = PTR_ERR(tfm);
+		luaL_error(L, "Failed to allocate SKCIPHER transform for %s (err %ld)", algname, err);
+	}
+	object->private = tfm;
```

Comment:
Actually you don't need to check for PTR_ERR on release, right? As you are only updating priv if it's ok..